### ISSUE TITLE
Base URL missing from redirect_uri when using OpenID Auth Provider

### ISSUE THREAD

**Issue by fenech** [alerta/alerta#1299]
**Issue Summary**

I am running Alerta behind a Traefik proxy, following the example shown in `docker-alerta/examples/traefik`.
The API is running with BASE_URL = "/alerta-api", and the web is at "/alerta". With AUTH_REQUIRED = False, I can go to `https://example.com/alerta/` and everything works as expected.

Now I am trying to enable authentication using OpenID (I am using ORY Hydra). When attempting to authenticate via OpenID, the request sent to the auth URL (defined via OIDC_AUTH_URL) is:

GET /oauth2/auth?response_type=code&client_id=alerta-client&redirect_uri=https://example.com&scope=openid+profile+email&display=popup&state=<state>

This causes the browser to open the login page inside a popup, as expected, but the `redirect_uri` query param does not include the base path, so when I log in, I am redirected to the main page of my application (within the popup) and the Alerta page is left stuck "Authenticating with OpenID ...".

**Environment**
- OS: Linux
- API version: 6.8.3
- Deployment: Docker

- Database: Postgres 12.4

- Server config:
  Auth enabled? Yes
  Auth provider? OpenID
  Customer views? No

- web UI version: 6.8.3
- CLI version: 6.8.3

**To Reproduce**
1. Go to https://example.com/alerta/
2. Enter login details and click login
3. The user is logged in and redirected to https://example.com/

**Expected behavior**
On successful login, popup closes and user is logged in and can access https://example.com/alerta/ in the main window.

**Comment by satterly** [comment]
The 'redirectUri' is sent by the web UI to the Alerta API and then forwarded to the OpenID Connect provider for authorization. It is auto-discovered using `window.location.origin`. See 
https://github.com/alerta/alerta-webui/blob/master/src/services/auth.ts#L95

If I use `window.location.href` instead of `window.location.origin` it includes the path in the redirect URL so it looks like this ... 

```
127.0.0.1 - - [25/Oct/2020 17:35:06] "GET /?response_type=code&client_id=736147134702-2lk9kqg6jueldtej10ajqo89jnbakr71.apps.googleusercontent.com&redirect_uri=http://local.alerta.io:8000/web/&scope=openid+profile+email&display=popup&state=rb5806j102 HTTP/1.1" 200 -
```

**Comment by satterly** [comment]
@fenech Can you provide a docker-compose.yaml file and ORY Hydra config that I can use to test this? Thanks.

**Comment by satterly** [comment]
If you only have Alerta on https://example.com then you can remove the `PathPrefix` in the traefik config and only register the domain (without the path) as a redirect URI.

```

  webui:
    build:
      context: webui
    networks:
      net:
        aliases:
          - web
    labels:
      - "traefik.backend=webui"
      - "traefik.http.routers.webui.rule=Host(`local.alerta.io`)"
      - "traefik.http.routers.webui.middlewares=webui-stripprefix@docker"
      - "traefik.http.middlewares.webui-stripprefix.stripprefix.prefixes=/web"

```

However, if you serve content on https://example.com as well as the Alerta web on https://example.com/alerta then this won't work.

**Comment by fenech** [comment]
Yeah, ours is the second case: we serve other content on the same domain, and use "Path" rules to control routing.

I'm working on a docker-compose file to reproduce the issue now!

**Comment by satterly** [comment]
I think alerta/alerta-webui#380 should fix the issue but it would be good to test before merging.

**Comment by fenech** [comment]
I was working on a self-contained docker-compose based on the use case in our platform but it was getting complicated...some of the images we use are in a private registry, and would require changes before they could be published elsewhere.

If you don't mind, I'll test that change in our stack and let you know how it goes. Sorry for the inconvenience.

**Comment by fenech** [comment]
I used this Dockerfile to build an `alerta-web` image:
```Dockerfile
FROM node:lts-alpine as build-stage

ARG BASE_URL=/alerta

RUN apk add --no-cache git
WORKDIR /app
RUN git clone --single-branch --branch openid-redirect-uri-base-path https://github.com/alerta/alerta-webui.git . \
  && npm install && BASE_URL="$BASE_URL" npm run build

FROM nginx:stable-alpine as production-stage
COPY --from=build-stage /app/dist /app
COPY config.json /app/config.json
COPY nginx.conf /etc/nginx/nginx.conf
EXPOSE 80
CMD ["nginx", "-g", "daemon off;"]
```

I can confirm that the request is made with the correct `redirect_uri`, in this case `/alerta/`
```
/oauth2/auth?client_id=alerta-client&consent_verifier=[ID]&display=popup
&redirect_uri=https%3A%2F%2Fexample.com%2Falerta%2F
&response_type=code&scope=openid+profile+email&state=[state]
```

So looks like the fix in [PR #380](https://github.com/alerta/alerta-webui/pull/380) is good, thanks!
